src/js/shared/Http/index.mjs

```diff
@@ -0,0 +1,123 @@
+let baseUri
+let tokenApiResolver
+let tokenApiReady = new Promise( resolve => tokenApiResolver = resolve )
+
+let authApiResolver
+let authApiReady = new Promise( resolve => authApiResolver = resolve )
+
+
+let proxyProvider = (resource, options) => Promise.resolve({ resource, options })
+
+const join = (...args) => args.map(a => (''+a).split('/')).flat().filter(a => a!=='').join('/').replace(':/', '://')
+
+function setEndpoint(endpoint) {
+    baseUri = endpoint
+}
+
+function onToken(callback) {
+    if (tokenApiResolver) {
+        tokenApiResolver(callback)
+        tokenApiResolver = null
+    }
+}
+
+function getToken() {
+    return tokenApiReady.then(tokenGetter => tokenGetter())
+}
+
+function onAuthorize(callback) {
+    if (authApiResolver) {
+        authApiResolver(callback)
+        authApiResolver = null
+    }
+}
+
+function getAuthorization(grants) {
+    return authApiReady.then(authGetter => authGetter(grants))
+}
+
+
+// this gets called syncrhonously by the SDK itself, if needed
+export function proxy(callback) {
+    proxyProvider = callback
+}
+
+function send(module, method, params, http = {}) {
+
+    let path = http.path || `/${module}/${method}`
+    const headers = http.headers || {}
+    const body = params && JSON.stringify(params) || null
+    let query = http.parameters || ''
+    
+    Object.entries(params).forEach(([name, value]) => {
+        const find = '${param.' + name + '}'
+        path = path.replace(find, value)
+        query = query.replace(find, value)
+        Object.keys(headers).forEach(header => {
+            headers[header] = headers[header].replace(find, value)
+        })
+    })
+
+    return new Promise((resolve, reject) => {
```

Comment:
might be better to rewrite this as an async function, a lot of nested `.then`s going on

Reply:
Stylistically, I agree.

However, I've been resisting the urge to use `async/await` because we haven't used them yet in the code. I just did a quick search and saw it's used in the new version of Lifecycle.ready() in the Core SDK. Have we tested that code on all of the browsers we care about?